### ISSUE TITLE
unlambda: false positive on builtins

### ISSUE THREAD

**Issue by mmcloughlin** [go-critic/go-critic#762]
```go
package main

import "fmt"

func main() {
	f := func(s []int) int { return len(s) }
	fmt.Println(f([]int{1, 2, 3}))
}
```

`gocritic` reports:

```
./builtins.go:6:7: unlambda: replace `func(s []int) int { return len(s) }` with `len`
```

However making the suggested change produces code that does not compile:

```
./builtins.go:6:7: use of builtin len not in function call
```

**Comment by Quasilyte** [comment]
Thank you for reporting this.
Fixed.

**Comment by mmcloughlin** [comment]
Wow that was quick!
